src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -0,0 +1,615 @@
+      auto replace_pt = find_string_key(key);
+      if (replace_pt != iter_end()) {
+        if (key < pivot) {  //left
+          auto mut_iter = left->iter_idx(replace_pt->get_index());
+          left->journal_leaf_update(mut_iter, key, value, left->maybe_get_delta_buffer());
+        } else if (key >= pivot) { //right
+          auto mut_iter = right->iter_idx(replace_pt->get_index() - left->get_node_size());
+          right->journal_leaf_update(mut_iter, key, value, right->maybe_get_delta_buffer());
+        }
+      } else {
+        auto insert_pt = string_lower_bound(key);
+        if (key < pivot) {  //left
+          auto mut_iter = left->iter_idx(insert_pt->get_index());
+          left->journal_leaf_insert(mut_iter, key, value, left->maybe_get_delta_buffer());
+        } else {
+          auto mut_iter = right->iter_idx(insert_pt->get_index() - left->get_node_size());
+          right->journal_leaf_insert(mut_iter, key, value, right->maybe_get_delta_buffer());
+        }
+      }
+      return oc.tm.dec_ref(oc.t, get_laddr())
+        .safe_then([tuple = std::move(tuple)] (auto ret) {
+        return insert_ret(
+               insert_ertr::ready_future_marker{},
+               mutation_result_t(mutation_status_t::SPLITTED, tuple, std::nullopt));
+      });
+    });
+  }
+}
+
+OMapLeafNode::rm_key_ret
+OMapLeafNode::rm_key(omap_context_t oc, const std::string &key)
+{
+  logger().debug("{}: {} : {}","OMapLeafNode",  __func__, key);
+  if(!is_pending()) {
+    auto mut =  oc.tm.get_mutable_extent(oc.t, this)->cast<OMapLeafNode>();
+    return mut->rm_key(oc, key);
+  }
+
+  auto rm_pt = find_string_key(key);
+  if (rm_pt != iter_end()) {
+    journal_leaf_remove(rm_pt, maybe_get_delta_buffer());
+    logger().debug(
+      "{}: removed {}->{} {}", __func__,
+      rm_pt->get_node_key(),
+      rm_pt->get_node_val(),
+      rm_pt->get_string_val());
+      if (extent_is_below_min()) {
+        return rm_key_ret(
+               rm_key_ertr::ready_future_marker{},
+               mutation_result_t(mutation_status_t::NEED_MERGE, std::nullopt,
+                                 this->cast<OMapNode>()));
+      } else {
+        return rm_key_ret(
+               rm_key_ertr::ready_future_marker{},
+               mutation_result_t(mutation_status_t::SUCCESS, std::nullopt, std::nullopt));
+      }
+  } else {
+    return rm_key_ret(
+           rm_key_ertr::ready_future_marker{},
+           mutation_result_t(mutation_status_t::FAIL, std::nullopt, std::nullopt));
+  }
+
+}
+
+OMapLeafNode::list_keys_ret
+OMapLeafNode::list_keys(omap_context_t oc, std::string &start, size_t max_result_size)
+{
+  logger().debug("{}: {}","OMapLeafNode",  __func__);
+  auto result = list_keys_result_t();
+  iterator  iter = start == "" ?  iter_begin() : string_lower_bound(start);
+  for (; iter != iter_end() && result.keys.size() <= max_result_size; iter++) {
+    result.keys.push_back(iter->get_node_val());
+  }
+  if (iter == iter_end())
```

Comment:
Can you add a comment to explain the behaviour when .size() is > max_result_size?